classes/common/class-checkout-modifications.php

```diff
@@ -217,4 +236,46 @@ public static function attach_order_note( $order_id ) {
 			__( 'Customer requested delivery time:', 'bring-fraktguiden-for-woocommerce' ) . " $time_slot"
 		);
 	}
+
+	/**
+	 * Bag on door consent
+	 */
+	public static function bag_on_door_consent() {
+		$current_shipping_method = WC()->session->get( 'chosen_shipping_methods' );
+		
+		if ( ! empty ( $current_shipping_method ) && $current_shipping_method[0] === 'bring_fraktguiden:3584' ) {
+			$name = __FUNCTION__;
+	
+			echo "<div id='$name'>";
+			woocommerce_form_field($name, array(
+				'type'      => 'checkbox',
+				'class'     => array('input-checkbox'),
+				'label'     => __( 'I agree to have my package delivered with bag on door', 'bring-fraktguiden-for-woocommerce' ),
+				'required'	=> true
+			), WC()->checkout->get_value($name));
+			echo '</div>';
+		}
+	}
+
+	/**
+	 * Save user selected bag on door value to order meta
+	 */
+	public static function bag_on_door_order_meta( $order_id ) {
+		$name = "bag_on_door_consent";
+
+		if ( ! empty( $_POST[$name] ) ) {
+			update_post_meta( $order_id, $name, $_POST[$name] );
+		}
+	}
+
+	/**
+	 * Display order related bag on door value in WC order admin page
+	 */
+	public static function bag_on_door_admin_value( $order ) {
+		$consent = get_post_meta( $order->get_id(), "bag_on_door_consent", true );
+
+		if ( $consent == 1 ) {
+			echo '<p><strong>Bag on door: </strong> Enabled</p>';
```

Comment:
Missing translation

Reply:
Also unescaped html. You should use esc_html_e() for this. Pro tip: Use [WordPress-Coding-Standards](https://github.com/WordPress/WordPress-Coding-Standards)

Reply:
Fixed both issues but couldn't get WPCS working on VSC. Maybe you can provide your rulesets?